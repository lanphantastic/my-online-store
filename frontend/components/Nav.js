import React from 'react';
import Link from 'next/link';
import NavStyles from './styles/NavStyles';
import User from './User';

const Nav = () => (
  <NavStyles>
    <User>
      {({data: { me }}) => {
        if(me) return <p>{me.name}</p>
        return <p>User</p>;
      }}
    </User>
    <Link href="/items">
      <a>Shop</a>
    </Link>
    <Link href="/sell">
      <a>sell</a>
    </Link>
    <Link href="signup">
      <a>Signup</a>
    </Link>
    <Link href="orders">
      <a>Orders</a>
    </Link>
    <Link href="/me">
      <a>Account</a>
    </Link>
  </NavStyles>
);

export default Nav;