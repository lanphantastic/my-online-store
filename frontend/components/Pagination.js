import React from 'react';
import PaginationStyles from './styles/PaginationStyles';

const Pagination = () => (
  <PaginationStyles>
    <p>Hi i'm pagination</p>
  </PaginationStyles>
);

export default Pagination;
