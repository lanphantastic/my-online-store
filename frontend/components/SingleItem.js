import React, { Component } from 'react';
import gql from 'graphql-tag';
import { Query } from 'react-apollo';
import styled from 'styled-components';

import Error from './ErrorMessage';

const SingleItemStyle = styled.div`
  max-width: 1200px;
  margin: 2rem auto;
  box-shadow: ${props => props.theme.bs};
  display: grid;
  grid-auto-columns: 1fr;
  grid-auto-flow: column;
  min-height: 800px;
`;

const SINGLE_ITEM_QUERY = gql`
  query SINGLE_ITEM_QUERY($id: ID!) {
    item(where: { id: $id }) {
      id
      title
      description
      largeImage
    }
  }
`;

class SingleItem extends Component {
  render() {
    return (
      <div>
        <p>Single item component {this.props.id}</p>
      </div>
    );
  }
}

export default SingleItem;
