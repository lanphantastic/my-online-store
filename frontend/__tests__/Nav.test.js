import React from 'react';
import { mount } from 'enzyme';
import wait from 'waait';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';

import Nav from '../components/Nav';
import { CURRENT_USER_QUERY } from '../components/User';
import { fakeUser, fakeCartItem } from '../lib/testUtils';

const notSignedInMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: {
      data: {
        me: null,
      },
    },
  },
];

const signInMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: {
      data: {
        me: fakeUser(),
      },
    },
  },
];
const signInMocksWithCartItems = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: {
      data: {
        me: {
          ...fakeUser(),
          cart: [fakeCartItem(), fakeCartItem(), fakeCartItem()]
        },
      },
    },
  },
];

describe('<Nav />', () => {
  it('renders a minimal nav when signed out', async () => {
    const wrapper = mount(
      <MockedProvider mocks={notSignedInMocks}>
        <Nav />
      </MockedProvider>,
    );
    await wait();
    wrapper.update();
    const nav = wrapper.find('ul[data-test="nav"]');
    // console.log(wrapper.debug());
    expect(nav.children().length).toBe(2);
    expect(nav.text()).toContain('ShopSign In');
  });

  it('renders full nav when signed in', async () => {
    const wrapper = mount(
      <MockedProvider mocks={signInMocks}>
        <Nav />
      </MockedProvider>,
    );
    await wait();
    wrapper.update();
    const nav = wrapper.find('ul[data-test="nav"]');
    // console.log(wrapper.debug());
    expect(nav.children().length).toBe(6);
    // Make sure it show 6 Links, which are the Shop, Sell, Orders, Account, Sign Out and My Cart.

  });

  it('renders the amount of items in the cart', async () => {
    const wrapper = mount(
      <MockedProvider mocks={signInMocksWithCartItems}>
        <Nav />
      </MockedProvider>
    )
    await wait();
    wrapper.update();
    const nav =wrapper.find('ul[data-test="nav"]');
  })

});
