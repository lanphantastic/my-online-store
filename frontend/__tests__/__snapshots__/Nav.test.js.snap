// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders the amount of items in the cart 1`] = `
Array [
  <CSSTransition
    className="count"
    classNames="count"
    in={true}
    key=".$9"
    onExited={[Function]}
    timeout={
      Object {
        "enter": 400,
        "exit": 400,
      }
    }
    unmountOnExit={true}
  >
    <Transition
      appear={false}
      className="count"
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot
        className="count"
      >
        <div
          className="count CartCount__Dot-sc-9jwa7x-1 eENeUA"
        >
          9
        </div>
      </CartCount__Dot>
    </Transition>
  </CSSTransition>,
  <Transition
    appear={false}
    className="count"
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={
      Object {
        "enter": 400,
        "exit": 400,
      }
    }
    unmountOnExit={true}
  >
    <CartCount__Dot
      className="count"
    >
      <div
        className="count CartCount__Dot-sc-9jwa7x-1 eENeUA"
      >
        9
      </div>
    </CartCount__Dot>
  </Transition>,
  <CartCount__Dot
    className="count"
  >
    <div
      className="count CartCount__Dot-sc-9jwa7x-1 eENeUA"
    >
      9
    </div>
  </CartCount__Dot>,
  <div
    className="count CartCount__Dot-sc-9jwa7x-1 eENeUA"
  >
    9
  </div>,
]
`;
